File: client-support/src/main/java/org/contractfirst/generator/client/internal/StatusCode.java
package org.contractfirst.generator.client.internal;

import java.util.Objects;

/**
 * Represents the status code of a response of an API operation. The status code can either be a real numeric HTTP status code or the "default" status
 * code, a placeholder for not otherwise specified status codes.
 */
public class StatusCode {

  public static final StatusCode DEFAULT = new StatusCode(null);

  private final Integer code;

  private StatusCode(Integer code) {
    this.code = code;
  }

  public static StatusCode of(int statusCode) {
    return new StatusCode(statusCode);
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    StatusCode that = (StatusCode) o;
    return Objects.equals(code, that.code);
  }

  @Override
  public int hashCode() {
    return Objects.hash(code);
  }

  @Override
  public String toString() {
    if (code == null) {
      return "default";
    } else {
      return code.toString();
    }
  }
}

File: client-support/src/main/kotlin/io/github/ruedigerk/contractfirst/generator/client/internal/StatusCode.kt
package io.github.ruedigerk.contractfirst.generator.client.internal

import java.util.*

/**
 * Represents the status code of a response of an API operation. The status code can either be a real numeric HTTP status code or the "default" status
 * code, a placeholder for not otherwise specified status codes.
 */
class StatusCode private constructor(private val code: Int?) {

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false
    
    val that = other as StatusCode
    
    return code == that.code
  }

  override fun hashCode(): Int {
    return Objects.hash(code)
  }

  override fun toString(): String {
    return code?.toString() ?: "default"
  }

  companion object {

    @JvmField
    val DEFAULT: StatusCode = StatusCode(null)

    @JvmStatic
    fun of(statusCode: Int): StatusCode {
      return StatusCode(statusCode)
    }
  }
}
